packages/babel-plugin-react-jsx/__tests__/TransformJSXToReactJSX-test.js

```diff
@@ -29,15 +30,307 @@ function transform(input, options) {
     }).code
   );
 }
 
 describe('transform react to jsx', () => {
+  it('throws error when sourceType is module and autoImport is require', () => {
+    let _error;
+    const code = `var x = <div><span /></div>`;
+    try {
+      transform(code, {
+        autoImport: 'require',
+      });
+    } catch (error) {
+      _error = error;
+    }
+    expect(_error).toEqual(
+      new SyntaxError(
+        'undefined: Babel `sourceType` must be set to `script` for autoImport ' +
+          'to use `require` syntax. See Babel `sourceType` for details.\n' +
+          codeFrame.codeFrameColumns(
+            code,
+            {start: {line: 1, column: 1}},
+            {highlightCode: true}
+          )
+      )
+    );
+  });
+
+  it('throws error when sourceType is script and autoImport is not require', () => {
+    let _error;
+    const code = `var x = <div><span /></div>`;
+    try {
+      transform(
+        code,
+        {
+          autoImport: 'namespace',
+        },
+        {sourceType: 'script'}
+      );
+    } catch (error) {
+      _error = error;
+    }
+    expect(_error).toEqual(
+      new SyntaxError(
+        'undefined: Babel `sourceType` must be set to `module` for autoImport ' +
+          'to use `namespace` syntax. See Babel `sourceType` for details.\n' +
+          codeFrame.codeFrameColumns(
+            code,
+            {start: {line: 1, column: 1}},
+            {highlightCode: true}
+          )
+      )
+    );
+  });
+
+  it("auto import that doesn't exist should throw error", () => {
+    let _error;
+    const code = `var x = <div><span /></div>`;
+    try {
+      transform(code, {
+        autoImport: 'foo',
+      });
+    } catch (error) {
+      _error = error;
+    }
+    expect(_error).toEqual(
+      new SyntaxError(
+        'undefined: autoImport must be one of the following: none, require, namespace, default, namedExports\n' +
+          codeFrame.codeFrameColumns(
+            code,
+            {start: {line: 1, column: 1}},
+            {highlightCode: true}
+          )
+      )
+    );
+  });
+
+  it('cannot use autoImport with createElement', () => {
+    let _error;
+    const code = `var x = <div><span /></div>`;
+    try {
+      transform(code, {
+        autoImport: 'namespace',
+        useCreateElement: true,
+      });
+    } catch (error) {
+      _error = error;
+    }
+    expect(_error).toEqual(
+      new SyntaxError(
+        'undefined: auto importing cannot be used with createElement. Consider setting ' +
+          '`useCreateElement` to false to use the new jsx function instead\n' +
+          codeFrame.codeFrameColumns(
+            code,
+            {start: {line: 1, column: 1}},
+            {highlightCode: true}
+          )
+      )
+    );
+  });
+
+  it('auto import can specify source', () => {
+    expect(
+      transform(`var x = <div><span /></div>`, {
+        autoImport: 'namespace',
+        importSource: 'foobar',
+      })
+    ).toMatchSnapshot();
```

Comment:
I wish we could use jest 24 so this could be `toMatchInlineSnapshot()` :( I'll try to fix it this week.